core/src/main/java/org/apache/james/core/healthcheck/Result.java

```diff
@@ -26,62 +26,27 @@
 public class Result {
 
     public static Result healthy(ComponentName componentName) {
-        return Builder.builder().componentName(componentName).status(ResultStatus.HEALTHY).build();
+        return new Result(componentName, ResultStatus.HEALTHY, Optional.empty(), Optional.empty());
     }
 
     public static Result unhealthy(ComponentName componentName, String cause) {
-        return Builder.builder().componentName(componentName).status(ResultStatus.UNHEALTHY).cause(cause).build();
+        return new Result(componentName, ResultStatus.UNHEALTHY, Optional.of(cause), Optional.empty());
     }
 
     public static Result unhealthy(ComponentName componentName, String cause, Throwable error) {
-        return Builder.builder().componentName(componentName).status(ResultStatus.UNHEALTHY).cause(cause).error(error).build();
+        return new Result(componentName, ResultStatus.UNHEALTHY, Optional.of(cause), Optional.of(error));
     }
 
     public static Result degraded(ComponentName componentName, String cause) {
```

Comment:
Can't we replace all these factory methods with the builder? We could add convenience `healthy`, `unhealthy()`, `unhealthy()` methods to it...

Reply:
This will bring us back to Matthieu concerns i guess.